lib/Geo/GeoJson.cpp

```diff
@@ -39,616 +41,765 @@
 #include "Basics/VelocyPackHelper.h"
 #include "Basics/debugging.h"
 #include "Geo/GeoParams.h"
-#include "Geo/S2/S2MultiPointRegion.h"
-#include "Geo/S2/S2MultiPolyline.h"
 #include "Geo/ShapeContainer.h"
+#include "Geo/S2/S2Points.h"
+#include "Geo/S2/S2Polylines.h"
 #include "Logger/Logger.h"
 
+namespace arangodb::geo::json {
 namespace {
-// Have one of these values:
-std::string const kTypeStringPoint = "Point";
-std::string const kTypeStringLineString = "LineString";
-std::string const kTypeStringPolygon = "Polygon";
-std::string const kTypeStringMultiPoint = "MultiPoint";
-std::string const kTypeStringMultiLineString = "MultiLineString";
-std::string const kTypeStringMultiPolygon = "MultiPolygon";
-std::string const kTypeStringGeometryCollection = "GeometryCollection";
-}  // namespace
 
-namespace {
-inline bool sameCharIgnoringCase(char a, char b) noexcept {
-  return arangodb::basics::StringUtils::toupper(a) ==
-         arangodb::basics::StringUtils::toupper(b);
+/// Cache for reuse memory between parsing few Region
+// TODO(MBkkt) It's unnecessary copy, we need to patch S2!
+//  But know we use cache.vertices to make single instead of N
+//  unnecessary copies when we parse N regions
+struct Cache {
+  std::vector<S2Point> vertices;
+};
+
+thread_local Cache cache;
+
+// The main idea is check in our CI 3 compile time branches:
+// 1. Validation true
+// 2. Validation false kIsMaintainer true
+// 3. Validation false kIsMaintainer false
+#if defined(ARANGODB_ENABLE_MAINTAINER_MODE) && defined(__linux__) && \
+    !defined(__aarch64__)
+constexpr bool kIsMaintainer = true;
+#else
+constexpr bool kIsMaintainer = false;
+#endif
+
+constexpr std::string_view kTypeStringPoint = "point";
+constexpr std::string_view kTypeStringPolygon = "polygon";
+constexpr std::string_view kTypeStringLineString = "linestring";
+constexpr std::string_view kTypeStringMultiPoint = "multipoint";
+constexpr std::string_view kTypeStringMultiPolygon = "multipolygon";
+constexpr std::string_view kTypeStringMultiLineString = "multilinestring";
+constexpr std::string_view kTypeStringGeometryCollection = "geometrycollection";
+
+// TODO(MBkkt) t as parameter looks better, but compiler produce strange error
```

Comment:
"but compiler produce strange error"
Could you please elaborate a bit more? At first glance "toString(t)" it works well  when called from template https://godbolt.org/z/hvMsna5dT

Reply:
Just not works, I described it, didn't I?
It just write it's not constant if I use it from another template

Probably compiler bug

Reply:
Just curious. What was the code to trigger the problem? You description gives not much hints )

Reply:
I couldn't understand, it was with clang 14 if you curious

Reply:
Ah I see. I've checked only latest version. Clang 15 works fine. And older gcc/msvc also compiles fine. Looks indeed like a bug in clang 14 and below. And they fixed it in the 15. Maybe add a TODO for removing that kludge once clang has a fix in older versions/we switch to 15?

Reply:
I will add todo to try it in future version but it's only stylistic difference